### PR TITLE
feature/#485 migrate job status update for scenario status to using submission entity

### LINKED ISSUE #485: Migrate Scenario Management elements to the Submission entity API
The code for Scenario Management elements handles the changes in submission statuses to be reflected in the UI (see `_get_submittable_status()`).

This should be sitting on the recently provided Submission entity API that Taipy Core exposes.

### PR DESCRIPTION
**Pull request by toan-quach**
Resolves #485.

# Changes:
- Removed updating scenario status based on job
- Added updating scenario status using Submission entity status
- Removed unneeded code

### CODE REVIEW

File: taipy/gui_core/_context.py [hunk 1]
```diff
@@ -165,50 +149,45 @@ def scenario_refresh(self, scenario_id: t.Optional[str]):
...
-                    _SubmissionStatus.CANCELED,
+                    SubmissionStatus.COMPLETED,
+                    SubmissionStatus.FAILED,
+                    SubmissionStatus.CANCELED,
                 ):
-                    self.client_jobs_by_submission.pop(sub_id, None)
+                    self.client_submission.pop(sub_id, None)
                 else:
-                    self.client_jobs_by_submission[sub_id] = sub_details.set_status(new_status)
+                    self.client_submission[sub_id] = sub_details.set_submission(new_status)
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py]: the parameter should be a submission

File: taipy/gui_core/_context.py [hunk 2]
```diff
@@ -165,50 +149,45 @@ def scenario_refresh(self, scenario_id: t.Optional[str]):
...
 
         except Exception as e:
-            _warn(f"Job ({job_id}) is not available", e)
+            _warn(f"Job ({submission_id}) is not available", e)
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py]: Job => Submission

File: taipy/gui_core/_context.py [hunk 3]
```diff
@@ -441,69 +420,23 @@ def submit_entity(self, state: State, id: str, payload: t.Dict[str, str]):
...
                     submission_fn = self.gui._get_user_function(submission_cb)
                     if callable(submission_fn):
-                        job_ids = [j.id for j in (jobs if isinstance(jobs, list) else [jobs])]
+                        submission_entity = core_get(jobs[0].submit_id if isinstance(jobs, list) else jobs.submit_id)
                         client_id = self.gui._get_client_id()
                         module_context = self.gui._get_locals_context()
-                        sub_id = jobs[0].submit_id if isinstance(jobs, list) else jobs.submit_id
                         with self.submissions_lock:
-                            self.client_jobs_by_submission[sub_id] = _SubmissionDetails(
+                            self.client_submission[submission_entity.submit_entity_id] = _SubmissionDetails(
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py]: is job.submit_id != submisssion_entity.id ?

File: taipy/gui_core/_context.py [hunk 4]
```diff
@@ -557,12 +490,6 @@ def data_node_adapter(self, data):
 
         return None
 
-    def get_jobs_list(self):
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py]: this is used by the job_selector component

File: taipy/gui_core/_context.py [hunk 5]
```diff
@@ -144,9 +130,7 @@ def process_event(self, event: Event):
                     )
             except Exception as e:
                 _warn(f"Access to sequence {event.entity_id} failed", e)
-        elif event.entity_type == EventEntityType.JOB:
```
Comment (FredLL-Avaiga) [inline taipy/gui_core/_context.py:147]: this is used by the job_selector component

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
I was hoping that we could wait for the addition of user data to the submit before changing this whole thing.
This would break the job_selector component

**Review by FredLL-Avaiga — DISMISSED** [review]

**Review by FredLL-Avaiga — APPROVED** [review]

File: taipy/gui_core/_context.py [hunk 6]
```diff
@@ -892,3 +842,12 @@ def select_id(self, state: State, id: str, payload: t.Dict[str, str]):
             state.assign(_GuiCoreContext._DATANODE_VIZ_DATA_ID_VAR, data_id)
         elif chart_id := data.get("chart_id"):
             state.assign(_GuiCoreContext._DATANODE_VIZ_DATA_CHART_ID_VAR, chart_id)
+
+
+# TODO remove when Submission is supported by Core API
```
Comment (jrobinAV) [inline taipy/gui_core/_context.py:847]: Noted!

https://github.com/Avaiga/taipy/issues/612

**Review by jrobinAV — APPROVED** [review]
There are not so much tests on the gui_core context, and we remove the only code that was tested... Too bad. 🤣🤣🤣 

It looks good to me.

**Pull request merged**